packages/webview_flutter/webview_flutter_wkwebview/lib/src/v4/src/webkit_webview_controller.dart

```diff
@@ -0,0 +1,354 @@
+      <WKWebsiteDataType>{
+        WKWebsiteDataType.memoryCache,
+        WKWebsiteDataType.diskCache,
+        WKWebsiteDataType.offlineWebApplicationCache,
+      },
+      DateTime.fromMillisecondsSinceEpoch(0),
+    );
+  }
+
+  @override
+  Future<void> clearLocalStorage() {
+    return _webView.configuration.websiteDataStore.removeDataOfTypes(
+      <WKWebsiteDataType>{WKWebsiteDataType.localStorage},
+      DateTime.fromMillisecondsSinceEpoch(0),
+    );
+  }
+
+  @override
+  Future<void> runJavaScript(String javaScript) async {
+    try {
+      await _webView.evaluateJavaScript(javaScript);
+    } on PlatformException catch (exception) {
+      // WebKit will throw an error when the type of the evaluated value is
+      // unsupported. This also goes for `null` and `undefined` on iOS 14+. For
+      // example, when running a void function. For ease of use, this specific
+      // error is ignored when no return value is expected.
+      if (exception.details is! NSError ||
+          exception.details.code !=
+              WKErrorCode.javaScriptResultTypeIsUnsupported) {
+        rethrow;
+      }
+    }
+  }
+
+  @override
+  Future<String> runJavaScriptReturningResult(String javaScript) async {
+    final Object? result = await _webView.evaluateJavaScript(javaScript);
+    if (result == null) {
+      throw ArgumentError(
+        'Result of JavaScript execution returned a `null` value. '
+        'Use `runJavascript` when expecting a null return value.',
+      );
+    }
+    return result.toString();
+  }
+
+  /// Controls whether inline playback of HTML5 videos is allowed.
+  Future<void> setAllowsInlineMediaPlayback(bool allow) {
+    return _webView.configuration.setAllowsInlineMediaPlayback(allow);
+  }
+
+  @override
+  Future<String?> getTitle() => _webView.getTitle();
+
+  @override
+  Future<void> scrollTo(int x, int y) {
+    return _webView.scrollView.setContentOffset(Point<double>(
+      x.toDouble(),
+      y.toDouble(),
+    ));
+  }
+
+  @override
+  Future<void> scrollBy(int x, int y) {
+    return _webView.scrollView.scrollBy(Point<double>(
+      x.toDouble(),
+      y.toDouble(),
+    ));
+  }
+
+  @override
+  Future<Point<int>> getScrollPosition() async {
+    final Point<double> offset = await _webView.scrollView.getContentOffset();
+    return Point<int>(offset.x.round(), offset.y.round());
+  }
+
+  // TODO(bparrishMines): This is unique to iOS. Override should be removed if
+  // this is removed from the platform interface before webview_flutter version
+  // 4.0.0.
+  @override
+  Future<void> enableGestureNavigation(bool enabled) {
+    return _webView.setAllowsBackForwardNavigationGestures(enabled);
+  }
+
+  @override
+  Future<void> setBackgroundColor(Color color) {
+    return Future.wait(<Future<void>>[
+      _webView.setOpaque(false),
+      _webView.setBackgroundColor(Colors.transparent),
+      _webView.scrollView.setBackgroundColor(color),
```

Comment:
Since we're doing things as separate asynchronous steps we should really do this one first to avoid the possibility of a flicker.